static-analysis/teammates-macker.xml

```diff
@@ -1,45 +1,399 @@
+                        </from>
+                    </allow>          
+                </deny>
+            </access-rule>
+        </ruleset>
+        <ruleset name="UiComponent: Pagedata is only used in Controller">
+            <access-rule>
+                <deny>
+                    <from class="${ui}.**" >
+                        <exclude class="${ui}.pagedata.**" /> <!-- allow dependency within the package -->
+                    </from>
+                    <to class="${ui}.pagedata.**" />
+                    <allow>
+                        <from>
+                            <include class="${ui}.controller.**" />
+     
+                            <!-- Violations that need fixing -->
+                            <include class="${common}.datatransfer.questions.FeedbackRankOptionsQuestionDetails" />
+                            <include class="${common}.datatransfer.questions.FeedbackRankRecipientsQuestionDetails" />
+                            <include class="${ui}.template.AdminAccountDetailsInstructorCourseListTableRow" />
+                            <include class="${ui}.template.AdminAccountDetailsStudentCourseListTableRow" />
+                            <include class="${ui}.template.AdminAccountManagementAccountTableRow" />
+                            <include class="${ui}.template.FeedbackSessionPublishButton" />
+                            <include class="${ui}.template.FeedbackSessionsAdditionalSettingsFormSegment" />
+                            <include class="${ui}.template.FeedbackSessionsForm" />
+                            <include class="${ui}.template.InstructorFeedbackSessionActions" />
+                            <include class="${ui}.template.StudentProfileEditBox" />
+                        </from>
+                    </allow>          
+                </deny>
+            </access-rule>
+        </ruleset>
+        <ruleset name="UiComponent: Controller should be self-contained">
+            <access-rule>
+                <deny>
+                    <from class="${ui}.**" />
+                    <to class="${ui}.controller.**" />
+                    <allow>
+                        <from class="${ui}.controller.**" /> <!-- allow dependency within the package -->
+                    </allow>     
+                </deny>
+            </access-rule>
+        </ruleset>
+        <ruleset name="UiComponent: Automation should be self-contained">
+            <access-rule>
+                <deny>
+                    <from class="${ui}.**" />
+                    <to class="${ui}.automated.**" />
+                    <allow>
+                        <from class="${ui}.automated.**" /> <!-- allow dependency within the package -->
+                    </allow>        
+                </deny>
+            </access-rule>
+        </ruleset>
+        <ruleset name="UiComponent: Automation should not have dependencies to anything within the package">
```

Comment:
Similarly here, `Automation` -> `Automated actions`.
Also, `within the UI component` instead of the package.